tests/Microsoft.Maui.IntegrationTests/BaseBuildTest.cs

```diff
@@ -0,0 +1,119 @@
+﻿
+namespace Microsoft.Maui.IntegrationTests
+{
+	public class BaseBuildTest
+	{
+		char[] invalidChars = { '{', '}', '(', ')', '$', ':', ';', '\"', '\'', ',', '=', '.', '-', };
+
+		public string TestName
+		{
+			get
+			{
+				var result = TestContext.CurrentContext.Test.Name;
+				foreach (var c in invalidChars.Concat(Path.GetInvalidPathChars().Concat(Path.GetInvalidFileNameChars())))
+				{
+					result = result.Replace(c, '_');
+				}
+				return result.Replace("_", string.Empty, StringComparison.OrdinalIgnoreCase);
+			}
+		}
+
+		public string TestDirectory => Path.Combine(TestEnvironment.GetTestDirectoryRoot(), TestName);
+
+		public string TestNuGetConfig => Path.Combine(TestEnvironment.GetTestDirectoryRoot(), "NuGet.config");
+
+		// Properties that ensure we don't use cached packages, and *only* the empty NuGet.config
+		protected string[] BuildProps => new[]
+		{
+			"RestoreNoCache=true",
+			//"GenerateAppxPackageOnBuild=true",
+			$"RestorePackagesPath={Path.Combine(TestEnvironment.GetTestDirectoryRoot(), "packages")}",
+			$"RestoreConfigFile={TestNuGetConfig}",
+			// Avoid iOS build warning as error on Windows: There is no available connection to the Mac. Task 'VerifyXcodeVersion' will not be executed
+			$"CustomBeforeMicrosoftCSharpTargets={Path.Combine(TestEnvironment.GetMauiDirectory(),"src", "Templates", "TemplateTestExtraTargets.targets")}",
+			//Try not restore dependencies of 6.0.10
+			$"DisableTransitiveFrameworkReferenceDownloads=true",
+		};
+
+
+		/// <summary>
+		/// Copy NuGet packages that would not and set up NuGet.config
```

Comment:
would not what?

Reply:
Oops, this should have said something along the lines of "copy packages that are not installed as part of the workload". I've updated this to also mention the relevant part of dotnet.cake where this was copied from.